upload/admin/model/extension/smailyforopencart/admin.php

```diff
@@ -31,82 +31,59 @@ public function uninstall() {
-        $this->load->model('setting/setting');
-        $settings = $this->model_setting_setting->getSetting('smaily_for_opencart');
-        // Credentials
-        $username = $settings['smaily_for_opencart_username'];
-        $password = $settings['smaily_for_opencart_password'];
-        $subdomain = $settings['smaily_for_opencart_subdomain'];
-        // Url.
-        $apiUrl = 'https://' . $subdomain . '.sendsmaily.net/api/' . $endpoint . '.php';
-        // Curl call.
-        $ch = curl_init();
-        curl_setopt($ch, CURLOPT_URL, $apiUrl);
-        curl_setopt($ch, CURLOPT_HEADER, 0);
-        curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
-        curl_setopt($ch, CURLOPT_HTTPAUTH, CURLAUTH_BASIC);
-        curl_setopt($ch, CURLOPT_USERPWD, "$username:$password");
-
-        // API call with GET request.
-        if ($method === 'GET') {
-            if (!empty($data)) {
-                curl_setopt($ch, CURLOPT_URL, $apiUrl . '?' . http_build_query($data));
-            }
-            $api_call = json_decode(curl_exec($ch), true);
-            // Response code from Smaily API.
-            $http_code = curl_getinfo($ch, CURLINFO_HTTP_CODE);
-        // If Method POST.
-        } elseif ($method === 'POST') {
-            curl_setopt($ch, CURLOPT_POST, 1);
-            curl_setopt($ch, CURLOPT_POSTFIELDS, http_build_query($data));
-            $api_call = json_decode(curl_exec($ch), true);
-            // Response code from Smaily API.
-            $http_code = curl_getinfo($ch, CURLINFO_HTTP_CODE);
-            // Validate response
-            if (!array_key_exists('code', $api_call)) {
-                $this->log->write('Something went wrong with Smaily request!');
-            }
-            if (isset($api_call['code']) && (int) $api_call['code'] !== 101) {
-                $this->log->write($api_call['message']);
-            }
+    public function normalizeSubdomain($subdomain) {
+        // First, try to parse as full URL.
+        // If that fails, try to parse as subdomain.sendsmaily.net.
+        // Last resort clean up subdomain and pass as is.
+        if (filter_var($subdomain, FILTER_VALIDATE_URL)) {
+        $url = parse_url($subdomain);
+        $parts = explode('.', $url['host']);
+        $subdomain = (count($parts) >= 3) ? $parts[0] : '';
         }
-        // Return response if success.
-        if ($http_code === 200) {
-            // POST.
-            if ($method === 'POST') {
-                if (array_key_exists('code', $api_call) && (int) $api_call['code'] === 101) {
-                    $response = $api_call;
-                }
-            // GET.
-            } else {
-                $response = $api_call;
-            }
-        } else {
-            // Log error.
-            $this->log->write('Error in smaily api call with code: ' . $http_code);
+        elseif (preg_match('/^[^\.]+\.sendsmaily\.net$/', $subdomain)) {
+        $parts = explode('.', $subdomain);
+        $subdomain = $parts[0];
```

Comment:
Formatting: indentation.

Reply:
Fixed with 8c27561
added `<rule ref="Generic.WhiteSpace.ScopeIndent">` to my ruleset as it didn't light up